### PR TITLE
Laser Pointers Use Pixel Params

### PR DESCRIPTION
**Pull request by ghost**
laser pointer

Why doesn't everything though?

### CODE REVIEW

**Comment by Xhuis** [comment]
:+1:

**Comment by Delimusca** [comment]
the age of the pixels is upon us :+1:

**Comment by optimumtact** [comment]
>why doesn't everything use something introduced recently

you meme

**Comment by WJohn** [comment]
:+1:

**Comment by CosmicScientist** [comment]
:+1:

**Comment by GunHog** [comment]
Lovely :+1:

**Comment by freerealestate** [comment]
love u

**Comment by Alek2ander** [comment]
Runtime still doesn't chase those. :cat:

**Comment by Dorsisdwarf** [comment]
:+1:

**Comment by RemieRichards** [comment]
```
if(!click_params || !click_params["icon-x"] || !click_params["icon-y"])
    return
```
Please.

**Comment by bustygoku** [comment]
@RemieRichards Like this? (second commit)

**Comment by RemieRichards** [comment]
@bustygoku sort of
looking at it, I'd invert that check, replacing it with:
```
if(click_params)
    if(click_params["icon-x"])
      I.pixel_x = text2num(click_params["icon-x"])
    if(click_params["icon-y"])
      I.pixel_x = text2num(click_params["icon-y"])
```
that way, if it fails to find click data, you still get a pointer effect in the middle of the tile, and some output messages, rather than everything just stopping.

**Comment by bustygoku** [comment]
Thank you. I'll add it to the laser pointer gun PR as well.

**Comment by bustygoku** [comment]
@RemieRichards This isn't working properly, what am I doing wrong here? I carried over the viewers() suggestion from the gun pointers PR as well.

**Comment by RemieRichards** [comment]
@bustygoku define, "isn't working properly"?

File: code/game/objects/items/devices/laserpointer.dm [hunk 1]
```diff
@@ -140,12 +140,19 @@
...
+	for(var/mob/M in viewers(7,targloc))
 		if(M.client)
 			showto.Add(M.client)
 	var/image/I = image('icons/obj/projectiles.dmi',targloc,pointer_icon_state,10)
-	I.pixel_x = target.pixel_x + rand(-5,5)
-	I.pixel_y = target.pixel_y + rand(-5,5)
+	var/list/click_params = params2list(params)
+	if(click_params)
+		if(click_params["icon-x"])
+			I.pixel_x = text2num(click_params["icon-x"])
```
Comment (RemieRichards) [inline code/game/objects/items/devices/laserpointer.dm]: you removed the ```-16```
I should have mentioned you needed to keep them...
sorry, I'm tired.

**Comment by bustygoku** [comment]
Well, it's putting the pointer on the tile to the east of the tile I'm clicking on, and not accurate at all.

File: code/game/objects/items/devices/laserpointer.dm (hunk 1 above)
Comment (bustygoku) [inline code/game/objects/items/devices/laserpointer.dm]: S'ok! I'll readd.

**Comment by bustygoku** [comment]
This is good to go now by the way.

**Comment by tuypo1** [comment]
oh how did I miss this I will have to give it a try

**Pull request merged**